comp/networkdeviceconfig/impl/config_test.go

```diff
@@ -120,7 +120,6 @@ network_device_config_management:
 			mockConfig := mock.NewFromYAML(t, tt.configYaml)
 			_, err := newConfig(mockConfig)
 			assert.NotNil(t, err)
-			assert.ErrorContains(t, err, tt.expectedErr)
```

Comment:
Could we validate the content of the error? expectedErr is unused now

Reply:
See conversation here: https://github.com/DataDog/datadog-agent/pull/40813#discussion_r2339682059. Switching the from viper's UnmarshalKey to using our own changes what the error message contains.